server/datastore/mysql/migrations/tables/20220818101352_ChangeSoftwareVendorWidth.go

```diff
@@ -0,0 +1,78 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220818101352, Down_20220818101352)
+}
+
+func Up_20220818101352(tx *sql.Tx) error {
+	logger.Info.Println("Increasing width of software.vendor...")
+
+	//-----------------
+	// Add temp column.
+	//-----------------
+	if _, err := tx.Exec(
+		`ALTER TABLE software ADD COLUMN vendor_wide varchar(114) NULL, ALGORITHM=INPLACE, LOCK=NONE`); err != nil {
+		return errors.Wrapf(err, "creating temp column for vendor")
+	}
+
+	//---------------------
+	// Add uniq constraint
+	//---------------------
+	if _, err := tx.Exec(
+		"ALTER TABLE software ADD constraint unq_name UNIQUE (name, version, source, `release`, vendor_wide, arch)"); err != nil {
+		return errors.Wrapf(err, "adding new uniquess constraint")
+	}
+
+	//------------------
+	// Update in batches
+	//------------------
+	const updateStmt = `UPDATE software SET vendor_wide = vendor WHERE vendor_wide IS NULL LIMIT 500`
+	for {
+		res, err := tx.Exec(updateStmt)
+		if err != nil {
+			return errors.Wrapf(err, "updating temp vendor column")
+		}
+		affected, err := res.RowsAffected()
+		if err != nil {
+			return errors.Wrapf(err, "updating temp vendor column")
+		}
+		if affected == 0 {
+			break
+		}
+	}
+
+	//----------------
+	// Drop old index
+	//----------------
+	if _, err := tx.Exec(`ALTER TABLE software DROP KEY name`); err != nil {
+		return errors.Wrapf(err, "dropping old index")
+	}
+
+	//----------------
+	// Drop old column
+	//----------------
+	if _, err := tx.Exec(`ALTER TABLE software DROP COLUMN vendor, ALGORITHM=INPLACE, LOCK=NONE`); err != nil {
+		return errors.Wrapf(err, "dropping old column")
+	}
```

Comment:
Can we really do this in place? Maybe in the next release to make sure it's not in use anymore?

Reply:
Sure - I'll rename this to `vendor_old` then

Reply:
Hm, should we leave this as `vendor` and start using `vendor_wide`? Not sure how much we can about making this backwards compatible, so up to you. Just a thought.

Reply:
Good thinking but I think this should be fine - currently we only set the `vendor` for RPM based distros, also we don't specify in our docs how long the `vendor` column is.